--- a/src/util.cpp
+++ b/src/util.cpp
@@ -33,6 +33,8 @@
 #include <gtk/gtk.h>
 #include <libgnome/gnome-url.h>
 
+#include <cstring>
+
 using namespace std;
 
 
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -81,7 +81,7 @@
 		GNOME_PARAM_POPT_TABLE, options,
 		GNOME_PARAM_APP_DATADIR, DATADIR,
 		NULL);
-    gnome_window_icon_set_default_from_file(GNOMEICONDIR "/" PACKAGE ".png");
+    gtk_window_set_default_icon_from_file(GNOMEICONDIR "/" PACKAGE ".png", NULL);
 
 
     // Argument parsing:
